fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -99,11 +104,46 @@ const socialSlice = createSlice({
         }
       },
     },
+
+    // Add reactions to a chirp
+    addReaction: {
+      prepare(laoId: Hash | string, reaction: ReactionState): any {
+        return { payload: { laoId: laoId.valueOf(), reaction: reaction } };
+      },
+      reducer(state, action: PayloadAction<{
+        laoId: string,
+        reaction: ReactionState,
+      }>) {
+        const { laoId, reaction } = action.payload;
+
+        if (!(laoId in state.byLaoId)) {
+          state.byLaoId[laoId] = {
+            allIdsInOrder: [],
+            byId: {},
+            byUser: {},
+            reactionsByChirp: {},
+          };
+        }
+
+        const store = state.byLaoId[laoId];
+
+        if (!store.reactionsByChirp[reaction.chirp_id]) {
+          store.reactionsByChirp[reaction.chirp_id] = { [reaction.codepoint]: [reaction.sender] };
+        } else if (!store.reactionsByChirp[reaction.chirp_id][reaction.codepoint]) {
+          store.reactionsByChirp[reaction.chirp_id][reaction.codepoint] = [reaction.sender];
+        } else if (!store.reactionsByChirp[reaction.chirp_id][reaction.codepoint]
+          .includes(reaction.sender)) {
+          store.reactionsByChirp[reaction.chirp_id][reaction.codepoint].push(reaction.sender);
+        } else {
+          console.warn('You already reacted to this reaction');
```

Comment:
I don't think it's necessary. What we could do to ensure that the user is informed of this is changing the style of the icon or of the counter. This way, the user knows it whiteout having a console message (which shows up only in the web console anyway).
I can do it in a separated PR though.

Reply:
Also, technically, it's "The sender already reacted to this reaction", not necessarily the user who parses this message🙂 
I wouldn't mind keeping the logging: it's useful to know that the message was processed.
Do however make it a "debug" message, this is definitely not a serious issue that one needs to be `warn`'d about